.github/actions/did-custom-action-code-change/src/main.js

```diff
@@ -0,0 +1,119 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+
+// When used, this requiredArgOptions will cause the action to error if a value has not been provided.
+const requiredArgOptions = {
+  required: true,
+  trimWhitespace: true
+};
+
+const fileInput = core.getInput('files-with-code') || '';
+const filesToCheckForChanges = fileInput && fileInput.length > 0 ? fileInput.split(',') : null;
+
+const folderInput = core.getInput('folders-with-code') || '';
+const foldersToCheckForChanges = folderInput && folderInput.length > 0 ? folderInput.split(',') : null;
+
+const prNumber = core.getInput('pr-number') || github.context.issue.number;
+const org = core.getInput('org') || github.context.repo.owner;
+const repo = core.getInput('repo') || github.context.repo.repo;
+
+const token = core.getInput('token', requiredArgOptions);
+const octokit = github.getOctokit(token);
+
+function getPrFolders(prFiles) {
+  const prFoldersSet = new Set();
+
+  if (prFiles) {
+    prFiles.forEach(f => {
+      const folder = f.substring(0, f.lastIndexOf('/'));
+      prFoldersSet.add(folder); // Adds the whole folder directory minus the file.ext
+
+      let baseFolder = '';
+      folder.split('/').forEach((fp, index) => {
+        prFoldersSet.add(fp);
+
+        baseFolder = index === 0 ? fp : `${baseFolder}/${fp}`;
+        prFoldersSet.add(baseFolder);
+      });
+    });
+  }
+
+  const prFolders = Array.from(prFoldersSet).sort((a, b) => (a > b ? 1 : b > a ? -1 : 0));
+  if (prFolders.length > 0) {
+    core.info(`\nThe following folders have changes in the PR:\n\t${prFolders.join('\n\t')}`);
+  } else {
+    core.info(`\nNo folders had changes in the PR.`);
+  }
+
+  return prFolders;
+}
+
+async function getPRFiles(org, repo, prNumber) {
+  let prFilesSet = new Set();
+  let prFiles = [];
```

Comment:
This is really nitpicking, but does this need to be declared at this scope?  You're only using this at the `.then`  for enumeration in favor of filling up `prFileSet`.

Reply:
It's not nitpicking.  You're right about the scope and I can change the the `let` for `prFilesSet` to a const.  I was trying different things out and that was a leftover.